x-pack/docs/en/security/securing-communications/update-tls-certificates.asciidoc

```diff
@@ -0,0 +1,784 @@
++
+[source,shell]
+----
+./bin/elasticsearch-certutil cert --in instances.yml --ca-cert ca/ca.crt \
+--ca-key ca/ca.key
+----
++
+[%collapsible%open]
+.Command parameters
+====
+   `-in`:: Specifies the name of the input YAML file (`instances.yml`)
+   containing information about each of your nodes.
+
+   `--ca-cert`:: Specifies the path to your new CA certificate (`ca.crt`) in PEM
+   format. You must also specify the `--ca-key` parameter.
+
+   `--ca-key`:: Specifies the path to the private key (`ca.key`) for your CA
+   certificate. You must also specify the `--ca-cert` parameter.
+====
+
+   a. Enter a name for the output file or accept the default of
+   `certificate-bundle.zip`.
+
+   b. When prompted, enter a password for each node certificate. It's good
+   practice to use a different password for each node certificate and store the
+   files in a secure location.
+
+   c. Decompress the `certificate-bundle.zip` file. The output contains a
+   directory for each of your nodes with a `<node-name>.p12` file.
+
+   d. Rename the `<node-name>.p12` file to match the existing file name
+   specified by the `xpack.security.transport.ssl.keystore.path` variable in 
+   the `elasticsearch.yml` file on each node.
+
+. [[start-rolling-restart-newca,step 3]]On the current node in your cluster where
+you're updating the keystore, start a
+<<restart-cluster-rolling,rolling restart>>.
++
+Stop at the step indicating *Perform any needed changes*, and then proceed to
+the next step in this procedure.
+
+. Replace your existing keystore with the new keystore, ensuring that the
+file names match. For example, `<node-name>.p12`.
++
+If your 
+<<cert-password-updates,keystore password is changing>>, then save the
+keystore with a new filename so that {es} doesn't attempt to reload the file
+before you update the password.
+
+. If you needed to save the new keystore with a new filename, update the
+`ES_PATH_CONF/elasticsearch.yml` file to use the filename of the new keystore.
+For example:
++
+[source,yaml]
+----
+xpack.security.transport.ssl.keystore.path: config/<node-name>.p12
+xpack.security.transport.ssl.keystore.type: PKCS12
+xpack.security.transport.ssl.truststore.path: config/elastic-stack-ca.p12
+xpack.security.transport.ssl.truststore.type: PKCS12
+----
+
+. Start the node where you updated the keystore, and then complete the remaining
+steps for a <<restart-cluster-rolling,rolling restart>>.
+
+. [[verify-keystore-newca,step 7]](Optional) Use the <<security-api-ssl,SSL certificate API>> to verify that
+{es} loaded the new keystore.
+
+. One node at a time, complete <<start-rolling-restart-newca>> through
+<<verify-keystore-newca>> until you've updated all keystores in your cluster.
+
+. (Optional) After rotating keystores on each node in your cluster, remove the
+old CA certificate from your truststore.
++
+[source,shell]
+----
+keytool -delete -noprompt -alias <old-ca>  -keystore config/elastic-stack-ca.p12 \
```

Comment:
We never told them which one is the old-ca and what it's alias is so I think users will be confused here. Maybe mention that if they used elasticsearch-certutil originally, the alias of the old CA would probably be `ca` and the type of the entry `PrivateKeyEntry` ? That doesn't cover all cases though so I'm not sure if this is helpful or more confusing

Reply:
When they import the `ca.crt` certificate into their existing truststore, users will provide an alias for the new CA certificate. So it's really just removing the one that isn't the new one.

Reply:
Sure, maybe we can explain this to them ? It might be obvious for us but I see a lot of folks looking at this confused as what is the old-ca

Reply:
I'm adding updating the previous step (revisions after this statement) to check that the new certificate was added to the truststore. We can then reference it at this point in the document so that users aren't surprised by what the old CA is or how to identify it.

> . Check that the new CA certificate was added to your truststore. 
> +
> [source,shell]
> ----
> keytool -keystore config/elastic-stack-ca.p12 -list
> ----
> When prompted, enter the password for the CA truststore.
> +
> The output should contain both the existing CA certificate and your new certificate. If you previously used the `elasticsearch-certutil` tool to generate your keystore, the alias of the old CA defaults to `ca` and the type of entry is `PrivateKeyEntry`.